pkg/apis/nmstate/v1/state_test.go

```diff
@@ -0,0 +1,103 @@
+package v1
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	yaml "github.com/ghodss/yaml"
+)
+
+var _ = Describe("NodeNetworkState", func() {
+	var (
+		obtainedStruct   NodeNetworkState
+		obtainedManifest []byte
+		desiredState     = State(`
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: up`)
+
+		currentState = State(`
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: down`)
+
+		NNSManifest = `
+apiVersion: nmstate.io/v1
+kind: NodeNetworkState
+metadata:
+  name: node01
+  creationTimestamp: "1970-01-01T00:00:00Z"
+spec:
```

Comment:
Should be

```yaml
apiVersion: nmstate.io/v1
kind: NodeNetworkState
metadata:
  name: node01
spec:
  managed: true
  nodeName: node01
  desiredState:
    interfaces:
    - name: eth1
      type: ethernet
      state: up
state:
  currentState:
    interfaces:
    - name: eth1
      type: ethernet
      state: down
```

See my comments on types

Reply:
I did remove state: on purpose we really need it ? commented at your comment.